### PR TITLE
Problem: negative coin amount error when query supply liquid of non BaseCoinUnit

### PR DESCRIPTION
**Pull request by mmsqe**
👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,7 @@
 - [#1088](https://github.com/crypto-org-chain/chain-main/pull/1088) Upgrade solomachine to `v0.1.4` and ibc-go to `v8.5.1`.
 * [#1091](https://github.com/crypto-org-chain/chain-main/pull/1091) Update cometbft to `0.38.13`, sdk to `v0.50.10` and memiavl to latest.
 - [#1091](https://github.com/crypto-org-chain/chain-main/pull/1091) Upgrade cometbft to v0.38.13, cosmos-sdk to `v0.50.10`.
+- [#1097](https://github.com/crypto-org-chain/chain-main/pull/1097) Avoid negative coin amount error when query supply liquid of non BaseCoinUnit.
```
Comment (mmsqe) [inline CHANGELOG.md:16]:
```suggestion
- [#1099](https://github.com/crypto-org-chain/chain-main/pull/1099) Avoid negative coin amount error when query supply liquid of non BaseCoinUnit.
```

File: x/supply/keeper/keeper.go [hunk 2]
```diff
@@ -92,7 +91,12 @@
 
 // GetTotalSupply returns the current total supply in the system
 func (k Keeper) GetTotalSupply(ctx sdk.Context) sdk.Coins {
-	return sdk.NewCoins(k.bankKeeper.GetSupply(ctx, config.BaseCoinUnit))
+	var totalSupply sdk.Coins
+	k.bankKeeper.IterateTotalSupply(ctx, func(coin sdk.Coin) bool {
+		totalSupply = totalSupply.Add(coin)
+		return false
+	})
+	return totalSupply
```
Comment (yihuang) [inline x/supply/keeper/keeper.go:99]: Doesn't return all total supplies of all coin denoms now?

Comment (mmsqe) [reply x/supply/keeper/keeper.go:99]: yes? you mean only return BaseCoinUnit?

Comment (yihuang) [reply x/supply/keeper/keeper.go:99]: It only returns basecro previously? Is there a bug inside the GetSupply method?

Comment (mmsqe) [reply x/supply/keeper/keeper.go:99]: GetSupply is for single denom, need switch to IterateTotalSupply if we need all denoms

Comment (yihuang) [reply x/supply/keeper/keeper.go:99]: is the negative amount a bug in the GetSupply method?
Is this PR a bug fix, or just change it from returning single denom to multiple ones?

Comment (mmsqe) [reply x/supply/keeper/keeper.go:99]: yes, it's bug in query supply liquid, the fix is change to return for all denoms

Comment (yihuang) [reply x/supply/keeper/keeper.go:99]: i see, because the other operands in the calculation contains all denoms?

Comment (mmsqe) [reply x/supply/keeper/keeper.go:99]: yea, when moduleAccount contains non BaseCoinUnit

**Review by yihuang — APPROVED** [review]

**Pull request merged**